pvlib/snow.py

```diff
@@ -185,3 +185,133 @@ def dc_loss_nrel(snow_coverage, num_strings):
+
+
+def loss_townsend(snow_total, snow_events, surface_tilt, relative_humidity,
+                  temp_air, poa_global, slant_height, lower_edge_drop_height,
+                  angle_of_repose=40):
+    '''
+    Calculates monthly snow loss based on the Townsend monthly snow loss
+    model [1]_.
+
+    Parameters
+    ----------
+    snow_total : array-like
+        Snow received each month. Referred to as S in [1]_ [cm]
+
+    snow_events : array-like
+        Number of snowfall events each month. Referred to as N in [1]_ [-]
+
+    surface_tilt : float
+        Tilt angle of the array [deg]
+
+    relative_humidity : array-like
+        Monthly average relative humidity [%]
+
+    temp_air : array-like
+        Monthly average ambient temperature [C]
+
+    poa_global : array-like
+        Monthly plane of array insolation [Wh/m2]
+
+    slant_height : float
+        Row length in the slanted plane of array dimension [m]
+
+    lower_edge_drop_height : float
+        Drop height from array edge to ground [m]
+
+    angle_of_repose : float, default 40
+        piled snow angle, assumed to stabilize at 40°, the midpoint of
+        25°-55° avalanching slope angles [deg]
+
+    Returns
+    -------
+    loss : array-like
+        Monthly average DC capacity loss fraction due to snow coverage
+
+    Notes
+    -----
+    This model has not been validated for tracking arrays; however, for
+    tracking arrays [1]_ suggests using the maximum rotation angle in place
+    of surface_tilt.
+
+    References
+    ----------
+    .. [1] Townsend, Tim & Powers, Loren. (2011). Photovoltaics and snow: An
+       update from two winters of measurements in the SIERRA. Conference
+       Record of the IEEE Photovoltaic Specialists Conference.
+       003231-003236. 10.1109/PVSC.2011.6186627.
+    '''
+
+    C1 = 5.7e04
+    C2 = 0.51
+
+    snow_total_prev = np.roll(snow_total, 1)
+    snow_events_prev = np.roll(snow_events, 1)
+
+    effective_snow = _townsend_effective_snow(snow_total, snow_events)
+    effective_snow_prev = _townsend_effective_snow(
+        snow_total_prev,
+        snow_events_prev
+    )
+    effective_snow_weighted = (
+        1 / 3 * effective_snow_prev
+        + 2 / 3 * effective_snow
+    )
+    effective_snow_weighted_m = effective_snow_weighted / 100
+
+    drop_height_clipped = np.maximum(lower_edge_drop_height, 0.01)
+    gamma = (
+        slant_height
+        * effective_snow_weighted_m
+        * cosd(surface_tilt)
+        / (drop_height_clipped**2 - effective_snow_weighted_m**2)
+        * 2
+        * tand(angle_of_repose)
+    )
+
+    ground_interference_term = 1 - C2 * np.exp(-gamma)
+    temp_air_kelvin = temp_air + 273.15
+    effective_snow_weighted_in = effective_snow_weighted / 2.54
+    poa_global_kWh = poa_global / 1000
+    loss_percentage = (
+        C1
+        * effective_snow_weighted_in
+        * cosd(surface_tilt)**2
+        * ground_interference_term
+        * relative_humidity
```

Comment:
More units... do we need to divide humidity by 100 to convert input units (percent) to the units of Eqn 3 (dimensionless fraction)?

Reply:
Oh yes, I think you're right. I've pushed a commit with that change. I've also adjusted the unit test from the original author so that it passes, but I don't know where the original author got the unit test's data. I think next it would be nice to add a unit test with data that reproduces one of the results in Table 2 of the paper.